src/inmanta/deploy/work.py

```diff
@@ -47,50 +47,41 @@ class TaskPriority(IntEnum):
     INTERVAL_REPAIR = 6
 
 
-@dataclass(frozen=True, kw_only=True)
-class PrioritizedTask(Generic[T]):
-    """
-    Resource action task with a priority attached. Lower values represent a higher priority. Negative priorities are reserved
-    for internal use.
-
-    This is a stateless representation of a task with a priority attached, unlike TaskQueueItem, which is a stateful element
-    in the task queue.
-    """
-
-    task: T
-    priority: TaskPriority
+Tiebreaker: typing.TypeAlias = int
 
 
 @functools.total_ordering
 @dataclass(kw_only=True)
 class TaskQueueItem:
     """
-    Task item for the task queue. Adds insert order field to each item and implements full ordering based purely on priority
-    and insert order (tiebreaker). Do not rely on == for task equality.
+    Task item for the task queue. Tasks are prioritized based on TaskPriority, with insert order as tiebreaker.
+
+    Implements full ordering based purely on priority and insert order (tiebreaker). Do not rely on == for task equality.
 
     The insert order field is a simple mechanic to maintain insert order for equal-priority tasks, as suggested in the
     heapq (collection type used by the agent queues) docs. The value should be monotonically rising within a queue. Gaps are
     allowed.
     """
 
-    task: PrioritizedTask[tasks.Task]
-    insert_order: int
+    task: tasks.Task
+    priority: TaskPriority
+    insert_order: Tiebreaker
 
     # Mutable state
     deleted: bool = False
 
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, TaskQueueItem):
             return NotImplemented
-        return (self.task.priority, self.insert_order) == (other.task.priority, other.insert_order)
+        return (self.priority, self.insert_order, self.deleted) == (other.priority, other.insert_order, other.deleted)
```

Comment:
Note to self: should this consider `self.task` as final tiebreaker value, purely for semantical reasons?